packages/firestore/src/local/indexeddb_mutation_queue.ts

```diff
@@ -161,34 +161,40 @@ export class IndexedDbMutationQueue implements MutationQueue {
     localWriteTime: Timestamp,
     mutations: Mutation[]
   ): PersistencePromise<MutationBatch> {
-    const batch = new MutationBatch(BATCHID_UNKNOWN, localWriteTime, mutations);
-    const dbBatch = this.serializer.toDbMutationBatch(this.userId, batch);
+    const documentStore = documentMutationsStore(transaction);
+    const mutationStore = mutationsStore(transaction);
 
-    return mutationsStore(transaction)
-      .add(dbBatch)
-      .next(batchId => {
-        batch.batchId = batchId;
-        this.documentKeysByBatchId[batchId] = batch.keys();
+    // The IndexedDb implementation in Chrome (and Firefox) does not handle
+    // compound indices with include auto-generated keys correctly. To ensure
```

Comment:
with include => "with included?" or "which include"?

(this comment is great, by the way. Thanks for including it)

Reply:
n/m you already fixed this. :)